Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -404,10 +404,18 @@ extension InternalRouter where Self: Router {
 extension Array where Element: MapboxDirections.Route {
     func index(mostSimilarTo route: Route) -> Int? {
         let target = route.description
-        return enumerated().min { (left, right) -> Bool in
-            let leftDistance = left.element.description.minimumEditDistance(to: target)
-            let rightDistance = right.element.description.minimumEditDistance(to: target)
-            return leftDistance < rightDistance
-        }?.offset
+        
+        guard let bestCandidate = map({
+            (route: $0, editDistance: $0.description.minimumEditDistance(to: target))
+        }).enumerated().min(by: { $0.element.editDistance < $1.element.editDistance }) else { return nil }
+
+
+        // If the most similar route is still more than 50% different from the original route,
```

Comment:
Why 50%?

Reply:
This number came from https://github.com/mapbox/mapbox-navigation-ios/issues/3597#issuecomment-982362158. There are only ever two road names in the route description per leg, so a 50% match is roughly one of the two names matching (modulo any differences in name length or minor things like “South”). We could probably lower the threshold without much problem, but this becomes a bugfarm, then the per-step approach in https://github.com/mapbox/mapbox-navigation-android/issues/3116#issuecomment-655710099 would be a decent plan B.